src/gui_common/charts/line/LineChart.cs

```diff
@@ -385,6 +395,9 @@ public void ClearDataSets()
 
             // Initialize line
             var dataLine = new DataLine((LineChartData)data.Value, data.Key == defaultDataSet, isChild ? 2 : 1);
+            dataLine.Connect("tree_exited", this, nameof(RemoveDataLineTooltipOnDeletion), new Array(dataLine));
+
+            dataLineTooltips.Add(dataLine, new List<ICustomToolTip>());
```

Comment:
For consistency and extra safety, add check for existing keys.
https://github.com/Revolutionary-Games/Thrive/blob/0f71e7936bbc2adef1bc1480c7418231202cb2a8/src/gui_common/charts/line/LineChart.cs#L391-L392